wordpress-mobile/WordPress-Android
# [Reader] Add comments and likes to Post Details Header

Fixes #19651

Add an "interaction counts" section containing Likes and Comments, the same that we show in the Feed post cards. The likes and comments in this section are clickable and send the user to the appropriate likers and comments screens.

A refactor in the Header Actions callbacks was done to avoid introducing multiple lambda parameters to some of the UI State mappers, so it would be nice to also test other Header actions (clicking Site, Follow, and Tags).


https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/bfbd7080-bd1a-4938-b013-ca8cb149d2a2

-----

## To Test:
1. Install and log into the Jetpack App
2. Open the Reader
3. Tap on any post
4. **Verify** the post details header shows the Likes and Comments count for the post
    1. If the post has no comments, only the likes are shown
    2. if the post has no likes, only the comments are shown
    3. if the post has no likes nor comments, the whole section is hidden
5. Tap the comment count
6. **Verify** the comment page is shown
7. Go back
8. Tap the like count
9. **Verify** the likers page is shown

Since some refactor was needed for the header callbacks, it would be nice to also test other Header actions: 
- Tap Site info
- Tap Follow/Unfollow button
- Tap Tag buttons

-----

## Regression Notes

1. Potential unintended areas of impact

    - Feed post card behavior changes.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing.

11. What automated tests I added (or what prevented me from doing so)

    - Updated unit tests for new Header Action callbacks
    - Added tests for Like and Comments callbacks

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
